pyomo/solvers/plugins/solvers/gurobi_direct.py

```diff
@@ -167,18 +167,23 @@ def _get_expr_from_pyomo_repn(self, repn, max_degree=2):
         if (degree is None) or (degree > max_degree):
             raise DegreeError('GurobiDirect does not support expressions of degree {0}.'.format(degree))
 
-        new_expr = repn.constant
-
-        for i,v in enumerate(repn.linear_vars):
-            new_expr += repn.linear_coefs[i] * self._pyomo_var_to_solver_var_map[v]
-            referenced_vars.add(v)
+        if len(repn.linear_vars) > 0:
+            list(map(referenced_vars.add, repn.linear_vars))
```

Comment:
`referenced_vars.update(repn.linear_vars)` seems like it would be equivalent, faster, and a bit easier to read

Reply:
Agreed.